catalog/app/containers/Bucket/PackageTree/PackageTree.tsx

```diff
@@ -435,18 +415,23 @@ function DirDisplay({
                           </Successors.Button>
                         )}
                         {actions.downloadPackage && (
-                          <Download.DownloadButton
+                          <Download.Button
                             className={classes.button}
-                            label={downloadLabel}
-                            onClick={openInDesktopState.confirm}
-                            path={downloadPath}
-                            selection={slt.selection}
-                          />
+                            label={
+                              !packageUri.path && slt.isEmpty ? 'Get package' : undefined
```

Comment:
does passing `undefined` here mean that there will be no label?
afair argument default won't be used if you are passing `undefined` explicitly

Reply:
Then intention here was to:
* use default label provided `Download.Button` in every case, except...
* if the path is "" (root of the package) then use "Get package"
  + but if the selection is not empty then still use the default label

I can write

```tsx
  <Download.Button
    {...(!packageUri.path && slt.isEmpty
      ? { label: 'Get package' }
      : {})}
  >
```
if it's less confusing.

---
  
When the property is not set or set to `undefined`, it will use default value during destructuring. Maybe, you confused with `null`?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring#default_value